internal/stream/stream.go

```diff
@@ -184,9 +187,38 @@ func (s *Stream) StartReader(reader Reader) {
 
 	sr.start()
 
-	for _, sm := range s.streamMedias {
+	for m, sm := range s.streamMedias {
 		for _, sf := range sm.formats {
 			sf.startReader(sr)
+			if m.Type == "video" {
+				cb := sf.runningReaders[sr]
+				if cb == nil {
+					continue
+				}
+
+				go func() {
+					for i, u := range s.CachedUnits {
+						var hasAU = false
+						switch tunit := u.(type) {
+						case *unit.H264:
+							hasAU = tunit.AU != nil
+						case *unit.H265:
+							hasAU = tunit.AU != nil
+						}
+						if hasAU {
+							cb(u)
+							// Wait 100ms for the first frame
+							if i == 0 {
+								time.Sleep(100 * time.Millisecond)
+							}
+							// Limit playback speed, to reduce the chance of packet loss
+							if i%5 == 0 {
+								time.Sleep(15 * time.Millisecond)
+							}
```

Comment:
You can try using a `time.Ticker` it takes into account how much time has passed instead of being a static delay -- unless you want the static delay.

Reply:
I really like the idea, I ended up doing `time.Sleep(time.Until(until))`.